feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/fee/utils/HydraDxQuoteSharedComputation.kt

```diff
@@ -0,0 +1,76 @@
+import io.novafoundation.nova.feature_swap_core.data.assetExchange.conversion.AssetExchangeQuoteArgs
+import io.novafoundation.nova.feature_swap_core.data.assetExchange.conversion.types.hydra.HydraDxAssetConversionFactory
+import io.novafoundation.nova.feature_swap_core.data.assetExchange.conversion.types.hydra.HydraDxSwapEdge
+import io.novafoundation.nova.feature_swap_core.domain.model.SwapDirection
+import io.novafoundation.nova.runtime.ethereum.StorageSharedRequestsBuilderFactory
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.FullChainAssetId
+import io.novasama.substrate_sdk_android.extensions.toHexString
+import io.novasama.substrate_sdk_android.runtime.AccountId
+import java.math.BigInteger
+import kotlinx.coroutines.CoroutineScope
+
+class HydraDxQuoteSharedComputation(
+    private val computationalCache: ComputationalCache,
+    private val assetConversionFactory: HydraDxAssetConversionFactory,
+    private val storageSharedRequestsBuilderFactory: StorageSharedRequestsBuilderFactory
+) {
+
+    suspend fun directions(
+        chain: Chain,
+        accountId: AccountId,
+        scope: CoroutineScope
+    ): Graph<FullChainAssetId, HydraDxSwapEdge> {
+        val key = "HydraDxDirections:${accountId.toHexString()}"
+
+        return computationalCache.useCache(key, scope) {
+            val assetConversion = getAssetConversion(chain, accountId, scope)
+            assetConversion.availableSwapDirections()
+        }
+    }
+
+    suspend fun quote(
+        chain: Chain,
+        accountId: AccountId,
+        fromAsset: Chain.Asset,
+        toAsset: Chain.Asset,
+        scope: CoroutineScope
+    ): AssetExchangeQuote {
+        val key = "HydraDxQuote:${accountId.toHexString()}"
+
+        return computationalCache.useCache(key, scope) {
+            val args = AssetExchangeQuoteArgs(
+                chainAssetIn = fromAsset,
+                chainAssetOut = toAsset,
+                amount = BigInteger.ZERO,
+                swapDirection = SwapDirection.SPECIFIED_IN
+            )
+
+            val assetConversion = getAssetConversion(chain, accountId, scope)
+            val swapDirections = directions(chain, accountId, scope)
+            val paths = assetConversion.getPaths(swapDirections, args)
+            assetConversion.quote(paths, args)
+        }
+    }
+
+    private suspend fun getAssetConversion(
+        chain: Chain,
+        accountId: AccountId,
+        scope: CoroutineScope
+    ): AssetConversion<HydraDxSwapEdge> {
+        val key = "HydraDxAssetConversion:${accountId.toHexString()}"
+
+        return computationalCache.useCache(key, scope) {
+            val storageSharedRequestBuilder = storageSharedRequestsBuilderFactory.create(chain.id)
+            val assetConversion = assetConversionFactory.create(chain)
+            assetConversion.runSubscriptions(accountId, storageSharedRequestBuilder)
```

Comment:
Have you tested that subscriptions actually work? We never call `storageSharedRequestBuilder.subscribe` so seems like dont actually subscribe atm. We can subscribe based on AggregateScope that CompotationalCache uses. In that case we pass it to the lambda in `useCache` in case its not already passed there

Reply:
Yes it doesn't work. I fixed it in [this](https://github.com/novasamatech/nova-wallet-android/pull/1633) pr